### ISSUE TITLE
[Quest] Firebase 9 compatibility

### ISSUE THREAD

**Issue by mikkopaderes** [mikkopaderes/ember-cloud-firestore-adapter#178]
Here's the game plan for this

- [x] Remove dependency with `ember-firebase-service` and use the compat SDK across the addon. See more details on it below.
- [x] Release a `v2.0.0` for the breaking changes
- [x] Slowly update the addon to use the modular SDK
- [x] Release `v3.0.0` when addon is 100% compat SDK free (timing might be on Firebase 10 release)

Notes:
- The game plan for this might change as development goes on
- By using `v2.0.0` you can slowly transition to use the modular SDK in your apps (see [here](https://firebase.google.com/docs/web/modular-upgrade#example_3_combining_version_8_and_version_9_code_styles)). But in my experience, these transition process may vary and is limited. For common and simple usage of this addon, you may not find anything that will be able to make use of this combination of v8 and v9 SDK API.

## Remove dependency with `ember-firebase-service`

Unfamiliar to some, `ember-firebase-service` exposes a service that represents [`firebase.app.App`](https://firebase.google.com/docs/reference/js/v8/firebase.app.App) which is heavily used in this addon. Moving forward, this addon will have its own `firebase` service but will marked as a private service like:

```javascript
import { inject as service } from '@ember/service';
import Controller from '@ember/controller';

import firebase from 'firebase/app';

export default class Application extends Controller {
  @service('-firebase')
  firebase;
}
```

In the future, the service will have no purpose in a pure Firebase modular SDK world, hence will be removed. This is the reason why I'll mark it as private, to discourage users from using it.

Users that are using the service should move away from it. Here's a sample migration path:

```javascript
import { inject as service } from '@ember/service';
import Controller from '@ember/controller';

import firebase from 'firebase/app';

export default class Application extends Controller {
  @service
  firebase;

  createTestRecord() {
    const db = this.firebase.firestore();

    db.collection('users').add({ ... });
  }
}
```

Instead, you can do:

```javascript
import Controller from '@ember/controller';

import { addDoc, collection, getFirestore } from 'firebase/firestore';

export default class Application extends Controller {
  createTestRecord() {
    const db = getFirestore();

    addDoc(collection(db, 'users'), { ... });
  }
}
```

Another small change is how the addon config env is being set up. Here's the migration path:

```javascript
let ENV = {
  firebase: {
    apiKey: '<api_key>',
    authDomain: '<auth_domain>',
    databaseURL: '<database_url>',
    projectId: '<project_id>',
    storageBucket: '<storage_bucket>',
    messagingSenderId: '<messaging_sender_id>'
  },

  'ember-cloud-firestore-adapter': {
    emulator: {
      hostname: 'localhost',
      firestorePort: 8080,
      authPort: 9099  // optional if not using auth
    },
  },
}
```

Will now be:

```javascript
let ENV = {
  'ember-cloud-firestore-adapter': {
    firebaseConfig: {
      apiKey: '<api_key>',
      authDomain: '<auth_domain>',
      databaseURL: '<database_url>',
      projectId: '<project_id>',
      storageBucket: '<storage_bucket>',
      messagingSenderId: '<messaging_sender_id>'
    },

    firestore: {
      settings: { ... },
      emulator: { ... }
    },

    auth: {
      emulator: { ... }
    }
  },
}
```

**Comment by mikkopaderes** [comment]
Updated the original post for a more detailed plan

**Comment by knownasilya** [comment]
Can this be closed?

**Comment by mikkopaderes** [comment]
> Can this be closed?

I originally plan to once Firebase 10 is released and I remove support for compat SDK. But technically, Firebase 9 is supported already in this addon so I'll try to re-review if this can be closed over the weekend.

**Comment by charlesfries** [comment]
#325 v10 support released in 4.1.1
